src/App/Fossa/FossaAPIV1.hs

```diff
@@ -580,3 +587,100 @@ resolveProjectDependencies apiOpts locator = fossaReq $ do
   (dependencies :: [ResolvedDependency]) <- responseBody <$> req GET (resolveProjectDependenciesEndpoint baseUrl locator) NoReqBody jsonResponse opts
 
   pure $ map unwrapResolvedDependency dependencies
+
+----------
+
+baseVsiUrl :: Url scheme -> Url scheme
+baseVsiUrl baseurl = baseurl /: "api" /: "proxy" /: "sherlock"
+
+data VSICreateScanRequestBody = VSICreateScanRequestBody
+  { vsiCreateScanRequestBodyOrgID :: Int
+  , vsiCreateScanRequestBodyProjectID :: Text
+  , vsiCreateScanRequestBodyRevisionID :: Text
+  }
+
+instance ToJSON VSICreateScanRequestBody where
+  toJSON VSICreateScanRequestBody{..} =
+    object
+      [ "OrganizationID" .= vsiCreateScanRequestBodyOrgID
+      , "ProjectID" .= vsiCreateScanRequestBodyProjectID
+      , "RevisionID" .= vsiCreateScanRequestBodyRevisionID
+      ]
+
+newtype VSICreateScanResponseBody = VSICreateScanResponseBody {unVSICreateScanResponseBody :: VSI.ScanID}
+
+instance FromJSON VSICreateScanResponseBody where
+  parseJSON = withObject "VSICreateScanResponseBody" $ \obj -> VSICreateScanResponseBody <$> obj .: "ScanID"
+
+vsiCreateScanEndpoint :: Url scheme -> Url scheme
+vsiCreateScanEndpoint baseurl = baseVsiUrl baseurl /: "scans"
+
+vsiCreateScan :: (Has (Lift IO) sig m, Has Diagnostics sig m) => ApiOpts -> ProjectRevision -> m VSI.ScanID
+vsiCreateScan apiOpts ProjectRevision{..} = fossaReq $ do
+  (baseUrl, baseOpts) <- useApiOpts apiOpts
+
+  Organization orgId _ <- getOrganization apiOpts
+  let projectID = renderLocator $ Locator "custom" projectName Nothing
+  let reqBody = VSICreateScanRequestBody orgId projectRevision projectID
+
+  (resBody :: VSICreateScanResponseBody) <- responseBody <$> req POST (vsiCreateScanEndpoint baseUrl) (ReqBodyJson reqBody) jsonResponse baseOpts
```

Comment:
Is this type annotation needed?  I think the newtype unwrapping function below hints the type well enough.